### PR TITLE
added methods for getting only changed root ids between timestamps

### PR DESCRIPTION
**Pull request by fcollman**
first cut at implementing this feature.. would like a sanity check from others before cleaning it up. not intended to be merged yet.

### CODE REVIEW

File: pychunkedgraph/backend/chunkedgraph.py [hunk 1]
```diff
@@ -1973,6 +1973,21 @@ def get_latest_roots(self, time_stamp: Optional[datetime.datetime] = get_max_tim
         return chunkedgraph_comp.get_latest_roots(self, time_stamp=time_stamp,
                                                   n_threads=n_threads)
 
+    def get_changed_roots(self,
```
Comment (sdorkenw) [inline pychunkedgraph/backend/chunkedgraph.py]: changed is a weird naming for this because ids never get _changed_
maybe get_root_id_changes()?

File: pychunkedgraph/backend/chunkedgraph_comp.py [hunk 2]
```diff
@@ -67,3 +98,47 @@ def get_latest_roots(cg,
...
+def get_changed_roots(cg,
+                      time_stamp_start: datetime.datetime,
+                      time_stamp_end: Optional[datetime.datetime] = None,
+                      n_threads: int = 1) -> Sequence[np.uint64]:
+
+    # Create filters: time and id range
+    max_seg_id = cg.get_max_seg_id(cg.root_chunk_id) + 1
+
+    n_blocks = np.min([n_threads * 3 + 1, max_seg_id])
+    seg_id_blocks = np.linspace(1, max_seg_id, n_blocks, dtype=np.uint64)
```
Comment (sdorkenw) [inline pychunkedgraph/backend/chunkedgraph_comp.py]: that certainly is an overkill for most timestamps
but its not easy to find a good range because I do not guarantee monotonicity of root ids in time

File: pychunkedgraph/backend/chunkedgraph.py (hunk 1 above)
Comment (fcollman) [reply pychunkedgraph/backend/chunkedgraph.py]: changed to get_delta_roots

File: pychunkedgraph/backend/chunkedgraph_comp.py (hunk 2 above)
Comment (fcollman) [reply pychunkedgraph/backend/chunkedgraph_comp.py]: added a feature to optionally specify a min_seg_id

**Comment by fcollman** [comment]
okay this is ready for review again... the remaining TODO would be to look for root_ids that were created and destroyed in the interim.  This isn't critical for my application but perhaps it should be done to make the interface as intuitive as possible... thoughts?

File: pychunkedgraph/backend/chunkedgraph_comp.py [hunk 3]
```diff
@@ -67,3 +104,49 @@ def get_latest_roots(cg,
...
+    new_root_ids = []
+    expired_root_id_candidates = []
+    for r1, r2 in results:
+        new_root_ids.extend(r1)
+        expired_root_id_candidates.extend(r2)
+    expired_root_id_candidates = np.array(expired_root_id_candidates, dtype=np.uint64)
+    expired_root_id_candidates = np.unique(expired_root_id_candidates)
+    rows = cg.read_node_id_rows(node_ids=expired_root_id_candidates,
+                                end_time=time_stamp_start)  
+    expired_root_ids = [k for (k, v) in rows.items()]
```
Comment (sdorkenw) [inline pychunkedgraph/backend/chunkedgraph_comp.py]: Why is this part necessary?

**Comment by sdorkenw** [comment]
@fcollman Looks good, I just do not understand these two lines at the end of `chunkedgraph_comp.get_delta_roots`

File: pychunkedgraph/backend/chunkedgraph_comp.py (hunk 3 above)
Comment (fcollman) [reply pychunkedgraph/backend/chunkedgraph_comp.py]: The reason for this is to filter out the root_ids that were created after the time stamp start, by looking for all new root_ids and going backward one FormerParent, the parallelized filter includes the root_ids that were created and then expired during the time interval, this filter removes those.  I forgot I had put this in.  I realized after your note that I needed a Child column filter on this call though, and I added comments.
